Add some nix-on-linux troubleshooting docs (#2250)
ComposableFi/composable: docs/docs/nix/troubleshooting.md

### 2. Permanent fix

To fix this issue permanently, we need to edit the `/etc/fstab` file using a text editor of your
choice.

Look for a line starting with `tmpfs` and update it like the following,  or append it to the bottom
if it doesn't exist yet.

```shell
$ tmpfs /tmp tmpfs rw,nodev,nosuid,size=16G 0 0
```

This change will permanently mount the `/tmp` directory with a fixed size of 16Gb.      

## Too many open files

> If you encounter a `Too many open files` error, we need to increase this limit.

Reviewer: ```suggestion
If you encounter a `Too many open files` error, this limit needs to be increased.
```
Ambiguous, could imply this fix needs to be done on our side.

The easiest way is to modify the `run` command with the `ulimit` command. Like the following:

```shell
$ bash -c 'ulimit -n 10000; nix run github:composablefi/composable#devnet-dali`
```

## Nix run command does not respect `nix.conf` configuration.

> If you're running nix on non-NixOS Linux, you may encounter some or all of the following issues, even if your
> config file(s) are set up correctly:

Reviewer: ```suggestion
If you're running Nix on a non-NixOS Linux, even if your config file(s) are set up correctly, 
you may encounter some or all of the following issues:
```
Nit style: Requirements should be grouped

> If you're running Nix on a non-NixOS Linux, even if your config file(s) are set up correctly, 

Author: ```suggestion
If you're running Nix on a non-NixOS Linux distribution, even if your config file(s) are set up correctly, 
```
